### PR TITLE
doc: Add Cilium container networking control flow

### PR DESCRIPTION
**Pull request by soumynathan**
This patch adds a control flow diagram that explains the cilium
container networking IP address Management with IPAM and with
default Kubernetes Cluster Node Pool CIDR.

Signed-off-by: Swaminathan Vasudevan <svasudevan@suse.com>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10387)

### CODE REVIEW

**Review by joestringer — CHANGES_REQUESTED** [review]
I provided some feedback on Slack. Reviewers should probably wait until those changes are made to provide additional insights.

**Comment by soumynathan** [comment]
Addressed review comments.

**Comment by joestringer** [comment]
test-docs-please

File: Documentation/concepts/ipam/index.rst [hunk 1]
```diff
@@ -12,6 +12,16 @@ Address Management
 
 Cilium supports multiple different address management modes:
 
+Cilium Container Networking Control Flow
```
Comment (joestringer) [inline Documentation/concepts/ipam/index.rst:16]: This paragraph has been inserted in between the sentence above which points out the different modes, and the toctree below which lists the modes with links. Did you mean to put this new section above or below the existing content?

**Comment by joestringer** [comment]
Docs test failed:
```
14:46:29  Please fix the following documentation warnings:
14:46:29  /srv/Documentation/concepts/ipam/index.rst:: WARNING: image file not readable: concepts/ipam/cilium_container_networking_control_flow.png
```

You may want to try using the local `make test-docs` target to validate this if you haven't already.

**Comment by soumynathan** [comment]
> Docs test failed:
> 
> ```
> 14:46:29  Please fix the following documentation warnings:
> 14:46:29  /srv/Documentation/concepts/ipam/index.rst:: WARNING: image file not readable: concepts/ipam/cilium_container_networking_control_flow.png
> ```
> 
> You may want to try using the local `make test-docs` target to validate this if you haven't already.

Sure will check it out.

File: Documentation/concepts/ipam/index.rst (hunk 1 above)
Comment (soumynathan) [reply Documentation/concepts/ipam/index.rst:16]: I have addressed your comments

**Comment by joestringer** [comment]
test-docs-please

**Review by joestringer — APPROVED** [review]

**Review by tgraf — APPROVED** [review]

**Comment by brb** [comment]
@soumynathan Too late to the party, but could you push a source code (?) for the png, so that we can update the diagram in the future?

**Comment by soumynathan** [comment]
@brb what change do you need to make, let me know I can churn it out.

**Comment by brb** [comment]
At the moment I don't have any change in mind, but if we need to do a change in the future, then the source code will be handy.

**Pull request merged**